lib/logstash/inputs/http_poller.rb

```diff
@@ -10,18 +10,36 @@
 # Reads from a list of urls and decodes the body of the response with a codec
 # The config should look like this:
 #
-#     input {
-#       http_poller {
-#         urls => {
-#           "test1" => "http://localhost:9200"
-#		        "test2" => "http://localhost:9200/_cluster/health"
-#         }
-#         request_timeout => 60
-#         interval => 10
-#       }
-#    }
+# input {
+#   http_poller {
+#     urls => {
+#       "test1" => "http://localhost:9200"
+#     "test2" => "http://localhost:9200/_cluster/health"
+#   }
+#   request_timeout => 60
+#   interval => 10
+#   codec => "json"
+# }
+# }
+#
+# # This plugin uses metadata, which by default is not serialized
+# # You'll need a filter like this to preserve the poller metadata
```

Comment:
Ah, good call. In fact that example is broken since it adds the metadata as stringified JSON. It's not actually possible to use the metadata due to a bug in `mutate` that makes it incapable of properly copying or renaming metadata.